tool/tsh/tsh.go

```diff
@@ -754,102 +761,106 @@ func Run(args []string, opts ...cliOption) error {
+	case command == kube.credentials.FullCommand():
 		err = kube.credentials.run(&cf)
-	case kube.ls.FullCommand():
+	case command == kube.ls.FullCommand():
 		err = kube.ls.run(&cf)
-	case kube.login.FullCommand():
+	case command == kube.login.FullCommand():
 		err = kube.login.run(&cf)
-	case kube.sessions.FullCommand():
+	case command == kube.sessions.FullCommand():
 		err = kube.sessions.run(&cf)
-	case kube.exec.FullCommand():
+	case command == kube.exec.FullCommand():
 		err = kube.exec.run(&cf)
-	case kube.join.FullCommand():
+	case command == kube.join.FullCommand():
 		err = kube.join.run(&cf)
 
-	case proxySSH.FullCommand():
+	case command == proxySSH.FullCommand():
 		err = onProxyCommandSSH(&cf)
-	case proxyDB.FullCommand():
+	case command == proxyDB.FullCommand():
 		err = onProxyCommandDB(&cf)
-	case proxyApp.FullCommand():
+	case command == proxyApp.FullCommand():
 		err = onProxyCommandApp(&cf)
 
-	case dbList.FullCommand():
+	case command == dbList.FullCommand():
 		err = onListDatabases(&cf)
-	case dbLogin.FullCommand():
+	case command == dbLogin.FullCommand():
 		err = onDatabaseLogin(&cf)
-	case dbLogout.FullCommand():
+	case command == dbLogout.FullCommand():
 		err = onDatabaseLogout(&cf)
-	case dbEnv.FullCommand():
+	case command == dbEnv.FullCommand():
 		err = onDatabaseEnv(&cf)
-	case dbConfig.FullCommand():
+	case command == dbConfig.FullCommand():
 		err = onDatabaseConfig(&cf)
-	case dbConnect.FullCommand():
+	case command == dbConnect.FullCommand():
 		err = onDatabaseConnect(&cf)
-	case environment.FullCommand():
+	case command == environment.FullCommand():
 		err = onEnvironment(&cf)
-	case mfa.ls.FullCommand():
+	case command == mfa.ls.FullCommand():
 		err = mfa.ls.run(&cf)
-	case mfa.add.FullCommand():
+	case command == mfa.add.FullCommand():
 		err = mfa.add.run(&cf)
-	case mfa.rm.FullCommand():
+	case command == mfa.rm.FullCommand():
 		err = mfa.rm.run(&cf)
-	case reqList.FullCommand():
+	case command == reqList.FullCommand():
 		err = onRequestList(&cf)
-	case reqShow.FullCommand():
+	case command == reqShow.FullCommand():
 		err = onRequestShow(&cf)
-	case reqCreate.FullCommand():
+	case command == reqCreate.FullCommand():
 		err = onRequestCreate(&cf)
-	case reqReview.FullCommand():
+	case command == reqReview.FullCommand():
 		err = onRequestReview(&cf)
-	case reqSearch.FullCommand():
+	case command == reqSearch.FullCommand():
 		err = onRequestSearch(&cf)
-	case config.FullCommand():
+	case command == config.FullCommand():
 		err = onConfig(&cf)
-	case configProxy.FullCommand():
+	case command == configProxy.FullCommand():
 		err = onConfigProxy(&cf)
-	case aws.FullCommand():
+	case command == aws.FullCommand():
 		err = onAWS(&cf)
-	case daemonStart.FullCommand():
+	case command == daemonStart.FullCommand():
 		err = onDaemonStart(&cf)
-	case f2Diag.FullCommand():
+	case command == f2Diag.FullCommand():
 		err = onFIDO2Diag(&cf)
+	case tid != nil && command == tid.ls.FullCommand():
```

Comment:
is changing every case in this giant switch really better than just adding a nil check to these two tid cases?

Reply:
An if would be saner, yes. Thanks for the suggestion.